web/src/muted_topics_ui.js

```diff
@@ -84,41 +100,40 @@ export function mute_topic(stream_id, topic, from_hotkey) {
     });
 }
 
-export function unmute_topic(stream_id, topic) {
-    // Accidentally unmuting a topic isn't as much an issue as accidentally muting
-    // a topic, so we don't show a popup after unmuting.
-    const data = {
-        stream_id,
-        topic,
-        op: "remove",
-    };
-
-    channel.patch({
-        url: "/json/users/me/subscriptions/muted_topics",
-        data,
-        success() {
-            feedback_widget.dismiss();
-        },
-    });
-}
-
 export function toggle_topic_mute(message) {
     const stream_id = message.stream_id;
     const topic = message.topic;
 
     if (user_topics.is_topic_muted(stream_id, topic)) {
-        unmute_topic(stream_id, topic);
+        update_topic_visibility_policy(
+            stream_id,
+            topic,
+            settings_config.topic_visibility_policy.UNMUTED,
+        );
     } else if (message.type === "stream") {
-        mute_topic(stream_id, topic, true);
+        update_topic_visibility_policy(
+            stream_id,
+            topic,
+            settings_config.topic_visibility_policy.MUTED,
+            true,
+        );
     }
 }
 
 export function mute_or_unmute_topic($elt, mute) {
```

Comment:
This function should be generalized to handle other visibility policies, in a follow-up PR.